src/components/pages/contact/contact-form.tsx

```diff
@@ -1,148 +1,165 @@
+import { Button } from '../../ui/buttons/button';
+import { SIMPLE_EMAIL_PATTERN } from '../../forms/constants';
 import styled from 'styled-components';
-import { SIMPLE_EMAIL_PATTERN } from '../../legacy/form/constants';
-import { CheckmarkIcon } from '../../legacy/icons/form-icons/checkmark';
-import { RightArrowIcon } from '../../legacy/icons/form-icons/right-arrow';
+import { TextStyles } from '../../typography';
+import { theme } from '../../layout/theme';
+import { Link } from '../../legacy/links/links';
+import { up } from '../../support/breakpoint';
 import { HoneypotField } from './honeypot';
 
-type RequestStatus = 'pending' | 'submitting' | 'success' | 'error';
 const API_ENDPOINT = '/api/contact-form';
 
-const StyledCaptionText = styled(CaptionText)`
-  color: inherit;
+const Grid = styled.div`
+  margin-bottom: 24px;
+  gap: 24px;
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+`;
+
+const Caption = styled.p`
+  ${TextStyles.textS}
+  margin: 16px 0 48px;
+`;
+
+const ErrorMessage = styled.p`
+  ${TextStyles.textXS}
+  font-weight: 700;
+  display: inline-block;
+
+  color: ${theme.palette.text.errorMessage};
+`;
+
+const StyledButton = styled(Button)`
+  margin-right: 16px;
+`;
+
+const GridItem = styled.div`
+  grid-column-start: span 12;
+
+  ${up('md')} {
+    grid-column-start: span 6;
+  }
 `;
+
+const FirstName = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextInput
+      name={'name'}
+      label={t('contact.name')}
+      control={control}
+      rules={{ required: t<string>('contact.error.name') }}
+    />
+  );
+};
+
+const Email = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextInput
+      name={'email'}
+      label={t('contact.email')}
+      control={control}
+      rules={{
+        required: t<string>('contact.error.email'),
+        pattern: {
+          value: SIMPLE_EMAIL_PATTERN,
+          message: t<string>('contact.error.email-undefined'),
+        },
+      }}
+    />
+  );
+};
+
+const MessageArea = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextArea
+      name={'message'}
+      label={t('contact.message')}
+      control={control}
+      rules={{ required: t<string>('contact.error.message') }}
+    />
+  );
+};
+
 interface FormData {
   name: string;
   email: string;
   message: string;
+
+  //Honeypot fields
+  firstName: '';
+  phone: '';
 }
 
-export const ContactForm: React.FC = () => {
-  const { t } = useTranslation();
-  const [requestStatus, setRequestStatus] = useState<RequestStatus>('pending');
+export const ContactForm = ({ onSuccess }: { onSuccess: () => any }) => {
   const {
-    register,
     handleSubmit,
-    formState: { errors },
-  } = useForm();
+    control,
+    formState: { isValid, isSubmitted },
+  } = useForm<FormData>({
+    defaultValues: {
+      name: '',
+      email: '',
+      message: '',
+      firstName: '',
+      phone: '',
+    },
```

Comment:
1. I think you can remove all those defaults as they are empty which is the default by itself, isn't it?
2. Even if you want to set the default please make sure to exclude the honeypot fields. Ideally those fields are not visible to the developer describing the form to prevent the noise coming from their declaration.

Reply:
> Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-> 
> components

The `defaultValues` were set to get rid of this warning. 
I removed the `defaultValues` now and added `value={field?.value || ''}` to `text-area.tsx` and `text-input.tsx`, which also prevents this warning

Reply:
Thank you!